### PR TITLE
IDEMPIERE-6336 - Adjust getList method in reflist

### PR DESCRIPTION
**Pull request by joseleite550**
https://idempiere.atlassian.net/browse/IDEMPIERE-6336

# Pull Request Checklist

- [ ] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [ ] My code follows the best practices of this project
- [ ] I have performed a self-review of my own code
- [ ] My code is easy to understand and review. 
- [ ] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [ ] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Please remove the migration scripts - as noted by @nmicoud 

The patch for MRefList looks OK - for example the WChosenboxListAssistant is showing inactive items

File: org.adempiere.base/src/org/compiere/model/MRefList.java [hunk 1]
```diff
@@ -303,7 +303,8 @@ else if ("V".equals(orderBy))
 			.append("FROM AD_Ref_List INNER JOIN AD_Ref_List_Trl trl ")
 			.append(" ON (AD_Ref_List.AD_Ref_List_ID=trl.AD_Ref_List_ID AND trl.AD_Language='")
 			.append(language).append("')");
-		sql.append(" WHERE AD_Ref_List.AD_Reference_ID=").append(AD_Reference_ID);
+		sql.append(" WHERE AD_Ref_List.AD_Reference_ID=").append(AD_Reference_ID)
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MRefList.java:306]: As noted by @nmicoud 
The change here must be consistent with https://wiki.idempiere.org/en/NF11_Enable_Showing_Inactive_Records_through_Reference

Comment (joseleite550) [reply org.adempiere.base/src/org/compiere/model/MRefList.java:306]: Ok

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Thanks @joseleite550 

can you please apply the following patch:
[pr2583PR0.patch.txt](https://github.com/user-attachments/files/18099862/pr2583PR0.patch.txt)

This is to show the inactive records with the ~ prefix/suffix.

**Review by CarlosRuiz-globalqss — APPROVED** [review]
tested

**Comment by joseleite550** [comment]
@CarlosRuiz-globalqss Could you take it to version 11?

**Comment by joseleite550** [comment]
it's ok

**Pull request merged**